packages/react/src/components/Coachmark/PositioningContainer/PositioningContainer.tsx

```diff
@@ -239,7 +239,7 @@ function useAutoDismissEvents(
 
     return () => events.dispose();
     // eslint-disable-next-line react-hooks/exhaustive-deps -- should only run on mount
-  }, []);
+  }, [dismissOnScroll]);
```

Comment:
I don't think this will work as desired either, since `dismissOnScroll` is re-created on every render right now. You could try wrapping it in `React.useCallback`, but that would require doing the same for `dismissOnLostFocus` and everything it uses (which might be fine depending on what the specific deps are and how often they mutate, or might not be practical).

Can you describe what you figured out that the root cause of the bug is? That will help me with reviewing and/or recommending possible solutions.

Reply:
The root cause of this bug has to do with dismissOnScroll not properly reading `positions`. In its current state `positions` is seen as undefined and `dismissOnLostFocus` is never called.